From 72514b8268b8caf2c91fafee1328faaf9930029e Mon Sep 17 00:00:00 2001
From: gee <unknown@unknown.com>
Date: Wed, 5 Aug 2015 17:08:18 +0100
Subject: [PATCH 02/20] PR54314: Include _ZTC* in libstdc++ exports
 [deprecated]

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54314
https://gcc.gnu.org/bugzilla/attachment.cgi?id=28065

Discussion:

1. gee 2012-08-21 13:38:57 UTC
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54314#c5

I think symbol _ZTCSt* need to be included
in libstdc++/config/abi/pre/gnu.ver so that
shared-library can export these symbols
unless user did append --disable-symvers.
nothing need to be done such as reverting
the commit or so.

2. Jason Merrill 2013-01-25 20:37:07 UTC
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54314#c19

This change turns out to be wrong; we don't want to
export the construction vtables, as they should only
be referenced by the VTTs which are emitted along
with them.  If other code is referring to those
symbols, that's the bug that should be fixed.

3. Kai Tietz 2013-09-10 16:19:48 UTC
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54314#c41

Add thunk _ZTv0_n12_NS* like in gnu.ver
---
 libstdc++-v3/config/abi/pre/gnu.ver | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libstdc++-v3/config/abi/pre/gnu.ver b/libstdc++-v3/config/abi/pre/gnu.ver
index d42cd37..7502fee 100644
--- a/libstdc++-v3/config/abi/pre/gnu.ver
+++ b/libstdc++-v3/config/abi/pre/gnu.ver
@@ -2064,7 +2064,7 @@ CXXABI_1.3.6 {
     __cxa_allocate_dependent_exception;
     __cxa_free_dependent_exception;
     __cxa_deleted_virtual;
-
+    _ZTC*;
 } CXXABI_1.3.5;
 
 CXXABI_1.3.7 {
-- 
2.5.0

